### PR TITLE
Js-Adagrams Cassy Archibald

### PR DESCRIPTION
**Pull request by cassyarchibald**
# JS Adagrams
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
--- | ---
What patterns were you able to use from your Ruby knowledge to apply to JavaScript? | It seemed like JavaScript had similar conditionals for ternary operators/if statement one liners. JavaScript also had the same methods for using objects/arrays. 
Did you need to use different strategies to find information online about JavaScript than you would for Ruby? | A little, sometimes I could not find an answer to my question in the documentation as I was trying to make a helper method from Ruby in JavaScript so I ended up in StackOverflow articles for those/in the MDN docs for data structure methods. 
What was a challenge you were able to overcome on this assignment? | In general figuring out how to translate the Ruby helper methods into manual JavaScript methods and how to split up my methods so they're not doing too many things.  
What is your favorite thing about learning a new programming language? | In general I think it's interesting to see what a new language is capable of/what quirks it has. 
What is your least favorite thing about learning a new programming language? | So many syntax errors, my muscle memory is still stuck in Ruby.  
Do you have any recommendations on how we could improve this project for the next cohort? | I don't think so, in general I enjoyed this assignment.

### CODE REVIEW

File: src/adagrams.js [hunk 1]
```diff
@@ -1,7 +1,361 @@
+// Helper Methods TODO Convert to arrow functions TODO
+// Returns random sample of array
+const sample = array => {
```
Comment (CheezItMan) [inline src/adagrams.js:3]: Ooooh!  Arrow helper functions!

File: src/adagrams.js [hunk 2]
```diff
@@ -1,7 +1,361 @@
...
+    case "Q":
+    case "Z":
+      return (score = 10);
+  }
+};
+//////// QUESTION  ////////
+//// How can I make an array of hashes (objects?) via map/the wordScore method from Adagrams?
+// My attempt is commented out below
+const wordScores = function wordScores(words) {
+  // Map through words to create array of hashes
```
Comment (CheezItMan) [inline src/adagrams.js:58]: I would put it like:

```javascript
return words.map( (word) => ({word: word, score: Adagrams.wordScore(word), letterCount: word.length });
```

Really good that you were looking at something like this.

**Comment by CheezItMan** [comment]
# JS Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions |   Check, you'll start to get JS muscle memory!  Don't worry
Small commits with meaningful commit messages |  Good number of commits and good commit messages
**Code Requirements** |
`drawLetters` method |  Check
Uses appropriate data structure to store the letter distribution | Check, nice use of `switch`
All tests for `drawLetters` pass | Check
`usesAvailableLetters` method |  Check
All tests for `usesAvailableLetters` pass |Check
`scoreWord` method | Check, see my notes on the `map` question.
Uses appropriate data structure to store the letter scores |Check
All tests for `scoreWord` pass |Check
`highestScoreFrom` method |Check
Appropriately handles edge cases for tie-breaking logic |Check
All tests for `highestScoreFrom` pass |Check
**Overall** |  Nicely done, I like the helper functions!
